relay-server/src/actors/spooler/mod.rs

```diff
@@ -719,12 +732,25 @@ impl BufferService {
         Ok(())
     }
 
+    async fn handle_health(&mut self, health: Health) -> Result<(), BufferError> {
+        match self.state {
+            // We do not check the ram, since we rely on `cache.envelope_buffer_size` to limit the
+            // memory usage and the number of the envelopes in the memory.
+            // Note: in the future we want to switch to `spool.envelopes.max_memory_size` option.
+            BufferState::Memory(_) | BufferState::MemoryFileStandby { .. } => health.0.send(true),
+            BufferState::Disk(ref disk) => health.0.send(disk.is_full().await.unwrap_or_default()),
```

Comment:
If boolean `true` means healthy, we should send `!disk.is_full()` here. Then the question is, use `unwrap_or_default()` or `unwrap_or(true)` (depends on the behavior)?

As discussed, an integration test would be good.

Reply:
@jjbayer  fixed and added integration test in [9419c93](https://github.com/getsentry/relay/pull/2113/commits/9419c938c33b4bfade867e04d6d2dceb8eaa2172)